src/services/EventService.ts

```diff
@@ -2,12 +2,43 @@ import { Event, AppUser, Attendance, RSVP } from '@Entities';
 import { AttendanceService, AttendanceServiceImpl } from './AttendanceService';
 import { RSVPService, RSVPServiceImpl } from './RSVPService';
 
-import { getRepository } from 'typeorm';
-import { MultipleAttendanceQuery } from '@Payloads';
+import { getRepository, FindManyOptions, Not } from 'typeorm';
+import { MultipleAttendanceQuery, MultipleEventQuery } from '@Payloads';
 
 export class EventService {
   constructor(private attendanceService: AttendanceService, private rsvpService: RSVPService) {}
 
+  private buildMultipleEventQuery(
+    multipleEventQuery: MultipleEventQuery,
+    canShowPending: boolean
```

Comment:
what about isOfficer here? then the logic for who should see pending is a little more clear - if you are an officer then you will be able to see xyz.

Another option that might be a little nicer would be to actually pass in the entire AppUser and do the logic of who can see what within the EventService for better encapsulation(not 100% sure on this one kinda up to you)

Reply:
I'm not sure if isOfficer is a good name, since there is no indication of what being an officer querying for multiple events entail. If we do use isOfficer, it is probably best to document what being an officer means in this context (what it allows an officer to see, as opposed to a non-officer member).

I agree with your second point, I think it would be nice to make the query builder function as generic as possible so too much logic is not being jammed into it

Reply:
To add on to your second point, I think it's best then to maybe not have the query builder function for GET multiple events. Can just jam the logic into the service function for GET multiple events or smth

Reply:
up to you - lmk what you decide on

Reply:
@godwinpang new commit pushed